File: 2079-watering-plants/2079-watering-plants.py
class Solution:
    def wateringPlants(self, plants: List[int], capacity: int) -> int:
        o=capacity
        cnt=0
        for i,j in enumerate(plants):
            if o<j:
                cnt+=2*i
                o=capacity
            cnt+=1
            o-=j 
        return cnt
        
        

File: 2079-watering-plants/2079-watering-plants.cpp
class Solution {
public:
    int wateringPlants(vector<int>& plants, int capacity) {
        int cnt=0,o=capacity;
        for(int i=0;i<plants.size();i++){
            if(o<plants[i]){
                cnt+=2*i;
                o=capacity;
            }
            cnt++;
            o-=plants[i];
        }
        return cnt;
        
    }
};